packages/amplify-cli-core/API.md

```diff
@@ -2034,6 +2138,33 @@ export interface Template {
 // @public (undocumented)
 export type TimedCodePath = ManuallyTimedCodePath | UntilExitTimedCodePath | FromStartupTimedCodePaths;
 
+declare namespace toolkitExtensions {
+    export {
+        isValidGraphQLAuthError,
+        handleValidGraphQLAuthError,
+        checkIfAuthExists,
+        confirmPrompt,
+        amplifyCLIConstants,
+        constructExeInfo,
+        copyBatch,
+        getAllCategoryPluginInfo,
+        getAllEnvs,
+        getAmplifyAppId,
+        getCategoryPluginInfo,
+        getEnvDetails,
+        getEnvInfo,
+        getFrontendPlugins,
+        getImportedAuthProperties,
+        getPluginInstance,
+        getPlugin,
+        getProjectConfig,
+        IAmplifyProjectDetails,
+        getProjectDetails,
+        getProjectMeta
+    }
```

Comment:
We should type those properties explicity, i.e. as `Function<X, Y, Z>` or `() => {}` somehow. So that we don't have to export things like `checkIfAuthExists` as global exports.

Reply:
Isn't it functionally the same thing though? If we export the function publicly in any part of the package, it is effectively the same thing as exporting it globally?

Reply:
It's not a full change but I hope it reflects what I mean here https://github.com/sobolk/amplify-cli/commit/dc10faec79bb0be0cdecf9d7e2db7724472a9148 .

I.e. if we're packing these things to toolkitExtensions. I shouldn't be able to write `import { checkIfAuthExists } from 'amplify-cli-core'`

Reply:
You can't do that with this change. You have to do:

```typescript
import { toolkitExtensions } from 'amplify-cli-core';

const { checkIfAuthExists } = toolkitExtensions;

```

Reply:
okay, that makes sense. Looks like API.md could present this better.